test/Microsoft.Framework.PackageManager.FunctionalTests/DnuListTests.cs

```diff
@@ -0,0 +1,216 @@
+using System.Linq;
+using Microsoft.Framework.FunctionalTestUtils;
+using Xunit;
+
+namespace Microsoft.Framework.PackageManager.FunctionalTests
+{
+    public class DnuListTests : IClassFixture<DnuTestContext>, IDisposable
+    {
+        private DnuTestContext _context;
+        private DnuListTestEnvironment _testEnvironment;
+
+        public DnuListTests(DnuTestContext context)
+        {
+            _context = context;
+            _testEnvironment = new DnuListTestEnvironment();
+
+            Console.WriteLine("Running test at {0}", _testEnvironment.RootDir);
+        }
+
+        public static IEnumerable<object[]> RuntimeComponents
+        {
+            get { return TestUtils.GetRuntimeComponentsCombinations(); }
+        }
+
+        public void Dispose()
+        {
+            _testEnvironment.Dispose();
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuList_EmptyProject_Default(string flavor, string os, string architecture)
+        {
+            var runtimeHomePath = _context.GetRuntimeHome(flavor, os, architecture);
+
+            var projectJson = Path.Combine(_testEnvironment.RootDir, "project.json");
+            File.WriteAllText(projectJson, @"{}");
+
+            string stdOut, stdErr;
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "list", "", out stdOut, out stdErr, environment: null, workingDir: _testEnvironment.RootDir));
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuList_EmptyProject_Details(string flavor, string os, string architecture)
+        {
+            var runtimeHomePath = _context.GetRuntimeHome(flavor, os, architecture);
+            var projectJson = Path.Combine(_testEnvironment.RootDir, "project.json");
+            File.WriteAllText(projectJson, @"{}");
+
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "list", "--details", environment: null, workingDir: _testEnvironment.RootDir));
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuList_SingleDependencyProject(string flavor, string os, string architecture)
+        {
+            var runtimeHomePath = _context.GetRuntimeHome(flavor, os, architecture);
+            var projectJson = Path.Combine(_testEnvironment.RootDir, "project.json");
+            File.WriteAllText(projectJson, @"{
+  ""dependencies"": {
+    ""alpha"": ""0.1.0""
+  },
+  ""frameworks"": {
+    ""dnx451"": {},
+    ""dnxcore50"": {}
+  }
+}");
+
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "restore", "--source " + _context.PackageSource, workingDir: _testEnvironment.RootDir));
+
+            string stdOut, stdErr;
+            Assert.Equal(0, DnuTestUtils.ExecDnu(runtimeHomePath, "list", "", out stdOut, out stdErr, environment: null, workingDir: _testEnvironment.RootDir));
+
+            var hits = stdOut.Split('\n').Where(line => line.Contains("* alpha 0.1.0"))
```

Comment:
Oh sure - but you could do `Assert.Collection(stdOut, first => { /* Do nothing */ }, second => Assert.Contains("* alpha"); third => ... );` Either that, or do an `Assert.True(2 == hits.Count(), $"Result: {stdOut}");`. The value is sort of not super important, but it would be useful to see the actual `stdOut` when your assertions fail.